### ISSUE TITLE
Error Updating CRM Record:  Method not found CrmServiceClient.set_EnableAffinityCookie

### ISSUE THREAD

**Issue by jmskarda** [MarkMpn/Sql4Cds#111]
Is UPDATE statement execution broken for anyone else?  

UPDATE statements used to work fine for me (months ago) but now I'm getting an error.  I know I am connected to the CRM because SELECT queries still work fine.  Sometimes errors like this occur because of incorrect/old DLL files in the GAC.  I don't want to waste time troubleshooting it if the issue not machine-specific.

```
UPDATE contact
SET 
  firstname = 'Bob'
WHERE
  contactid = 'aaaaaaaaa-0000-e000-000f-00000000a5e26' 
```

```
Method not found: 'Void Microsoft.Xrm.Tooling.Connector.CrmServiceClient.set_EnableAffinityCookie(Boolean)'.
Method not found: 'Void Microsoft.Xrm.Tooling.Connector.CrmServiceClient.set_EnableAffinityCookie(Boolean)'.
See the Execution Plan tab for details of where this error occurred
```

---
- Deployment: Online
- DB Version: 9.2.21064.148
- Connection Controls Version: 1.2021.6.44
- XrmToolBox Version: 1.2021.6.51
- Tool Version: 5.2.1.0

**Comment by MarkMpn** [comment]
This looks like the same issue as #102 - please make sure you’re on the latest version of XrmToolBox

**Comment by jmskarda** [comment]
Thank you.   I have confirmed that I am on the latest version of XrmToolBox (1.2021.6.51).  To be sure of this, I downloaded it again today and ran it from a different folder.   I also tried a fresh XRMToolbox install onto a VM that I use.  I get the same error on both machines.

However, I tried it again from a third machine (my personal machine at home) and it works fine.  

I only get the error on machines that I use for work.  They are probably more up to date with the most recent OS and other patches from Microsoft.  They also have some older D365 SDK DLL files installed into the GAC.  I don't even have Visual Studio installed on my home machine, let alone anything in the GAC.  If those older D365 SDK DLL files are causing the problem, then most people probably won't experience this issue.

I will investigate further and report back if I learn anything.

**Comment by jmskarda** [comment]
I resolved this by removing the Microsoft.Xrm.Tooling.Connector from my GAC. 

From a Visual Studio command prompt:
`gacutil /u Microsoft.Xrm.Tooling.Connector`

I had two versions installed (2 and 4). It removed both.  I'll eventually find out whether I actually needed them.  It's an old project that I inherited.

I suppose this means it was not a bug.  I would have thought that XrmToolbox and SQL4CDS would be self-contained.  Apparently they can be affected by the GAC.
